src/kprofile.cpp

```diff
@@ -5,6 +5,27 @@
 #include <QMessageBox>
 #include <QWebEngineSettings>
 
+QString askForSaveFilePath(const QString& suggestedName)
+{
+    const auto app = KiwixApp::instance();
+    const QString suggestedPath = app->getPrevSaveDir() + "/" + suggestedName;
+    const QString extension = suggestedName.section('.', -1);
+    const QString filter = extension != '.' ? "(*" + extension + ")" : "";
+
+    QString fileName = QFileDialog::getSaveFileName(
+            app->getMainWindow(), gt("save-file-as-window-title"),
+            suggestedPath, filter);
+
+    if (fileName.isEmpty())
+        return QString();
+    
+    if (!fileName.endsWith(extension)) {
+        fileName.append(extension);
+    }
```

Comment:
Above code considers a special case when `extension == "."`. If that case is really important then it has to be taken into account here as well.

Reply:
No longer needed since fileName cannot be empty due to check before.